Admonish breaks if `on_failure = "bail"` is set and `[preprocessor.admonish.default]` exists in `book.toml`

Admonish will fail and break the build entirely if `on_failure = "bail"` is set and  `[preprocessor.admonish.default]` exists, for example as documented at https://tommilligan.github.io/mdbook-admonish/reference.html#booktoml-configuration.

```toml
# book.toml
[preprocessor.admonish]
assets_version = "3.0.2" # do not edit: managed by `mdbook-admonish install`
command = "mdbook-admonish"
on_failure = "bail"

[preprocessor.admonish.default]
# collapsible = true # With or without this or any other value.

[preprocessor.admonish.renderer.test] # With or without this.
render_mode = "strip"
```

results in

```ini
[2024-06-18T01:41:58Z ERROR mdbook_admonish] Fatal error: values must be emitted before tables
[2024-06-18T01:41:58Z ERROR mdbook_admonish]   - values must be emitted before tables
2024-06-18 01:41:58 [ERROR] (mdbook::utils): Error: The "admonish" preprocessor exited unsuccessfully with exit status: 1 status
```

commenting out `on_failure` successfully completes with no warnings or errors, and values take effect with no obvious issue. There is no error message I can find in the rendered output or anywhere else, as the documentation says there should be when there is an error: "If rendering to HTML, an error message will be displayed in the book output."

It makes no difference whether any admonish blocks actually even exist in the mdbook.

This is the only configuration item for which I have observed this. It's a pretty annoying problem that having any defaults and [reporting invalid blocks](https://tommilligan.github.io/mdbook-admonish/overview.html?highlight=loud#invalid-blocks) are mutually exclusive.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -41,10 +41,6 @@ regex = "1.9.6"
 semver = "1.0.19"
 serde = { version = "1.0.188", features = ["derive"] }
 serde_json = "1.0.107"
-# Peer dependency of mdbook
-# The version of toml that mdbook uses internally (and uses in it's public api)
-# Only used for compatilibilty with the mdbook public api
-toml_mdbook = { package = "toml", version = "0.5.11" }
 toml = "0.8.1"
 toml_edit = { version = "0.22.13", optional = true }
 hex_color = { version = "3.0.0", features = ["serde"] }

--- a/src/bin/mdbook-admonish.rs
+++ b/src/bin/mdbook-admonish.rs
@@ -125,7 +125,7 @@ struct Preprocessors {
 
 /// Load the plugin specific config as a toml string, for private deserialization.
 fn admonish_config_string(config: &Config) -> Result<String> {
-    Ok(toml_mdbook::to_string(
+    Ok(toml::to_string(
         &config
             .preprocessor
             .admonish

--- a/src/book_config.rs
+++ b/src/book_config.rs
@@ -11,11 +11,12 @@ use crate::types::{AdmonitionDefaults, BuiltinDirective, BuiltinDirectiveConfig}
 /// Roundtrips config to string, to avoid linking the plugin's internal version of toml
 /// to the one publically exposed by the mdbook library.
 pub(crate) fn admonish_config_from_context(ctx: &PreprocessorContext) -> Result<Config> {
-    let table: String = toml_mdbook::to_string(
+    let table: String = toml::to_string(
         ctx.config
             .get_preprocessor("admonish")
             .context("No configuration for mdbook-admonish in book.toml")?,
-    )?;
+    )
+    .context("Could not serialize mdbook-admonish config. This is a bug in the toml library.")?;
     admonish_config_from_str(&table)
 }
 

